scripts/cluster-upgrade.sh

```diff
@@ -0,0 +1,161 @@
+        OCP_CURRENT_VERSION=`oc -n $CD_NAMESPACE get clusterdeployment $CD_NAME -o json | jq -r '.status.clusterVersionStatus.history[] | select(.state == "Completed") | .version' | head -n1`
+
+        SS_NAME=$CD_NAME-upgrade
+        SS_VERSION_FROM=`oc -n $CD_NAMESPACE get syncset $SS_NAME --show-labels --no-headers 2> /dev/null | sed 's|.*managed.openshift.io/from=\([^,]*\).*|\1|g'`
+        SS_VERSION_TO=`oc -n $CD_NAMESPACE get syncset $SS_NAME --show-labels --no-headers 2> /dev/null | sed 's|.*managed.openshift.io/to=\([^,]*\).*|\1|g'`
+        SS_CREATED_ON=$(date --date="$(oc -n $CD_NAMESPACE get syncset $SS_NAME --template '{{ .metadata.creationTimestamp }}' 2>/dev/null)" +"%s" 2>/dev/null)
+        SS_AGE_MIN=$(($(($NOW_EPOCH-$SS_CREATED_ON))/60))
+
+        # always report status
+        echo "done ($OCP_CURRENT_VERSION)"
+
+        if [ ! -z $OCP_VERSION_FROM ];
+        then
+            # upgrade
+
+            # do nothing if the current version in SS is what we got out of CD
+            if [ "$OCP_CURRENT_VERSION" == "$SS_VERSION_FROM" ];
+            then
+                UPGRADE_PROGRESSING="${UPGRADE_PROGRESSING}  $CD_NAME ($SS_VERSION_FROM->$SS_VERSION_TO) [age=$SS_AGE_MIN(min)]\n"
+                continue
+            fi
+
+            if [ "$OCP_CURRENT_VERSION" == "$OCP_VERSION_FROM" ];
+            then
+                SS_FILENAME=$TMP_DIR/$SS_NAME.syncset.yaml
+                cat << EOF > $SS_FILENAME
+apiVersion: hive.openshift.io/v1alpha1
+kind: SyncSet
+metadata:
+  name: $SS_NAME
+  namespace: $CD_NAMESPACE
+  labels:
+    managed.openshift.io/from: "$OCP_CURRENT_VERSION"
+    managed.openshift.io/to: "$OCP_VERSION_TO"
+spec:
+  clusterDeploymentRefs:
+  - name: $CD_NAME
+  resourceApplyMode: Sync
+  patches:
+  - apiVersion: config.openshift.io/v1
+    applyMode: ApplyOnce
+    kind: ClusterVersion
+    name: version
+    patch: '$OCP_PATCH'
+    patchType: merge
+EOF
+
+                oc -n $CD_NAMESPACE create -f $SS_FILENAME || oc -n $CD_NAMESPACE replace -f $SS_FILENAME
+                UPGRADE_STARTED="${UPGRADE_STARTED}  $CD_NAME ($OCP_VERSION_FROM->$OCP_VERSION_TO)\n"
+            elif [ -z $SS_VERSION_FROM ] && [ "$OCP_CURRENT_VERSION" != "$OCP_VERSION_TO" ];
+            then
+                # report can't upgrade if version is not upgradable and upgrade is not in progress
+                UPGRADE_NOT_POSSIBLE="${UPGRADE_NOT_POSSIBLE}  $CD_NAME ($OCP_CURRENT_VERSION)\n"
+            else
+                # cluster is already upgraded?
+                if [ -z $SS_VERSION_FROM ];
+                then
+                    # delete the syncset, it's not needed anymore
+                    oc -n $CD_NAMESPACE delete SyncSet $SS_NAME
+                    UPGRADE_DONE="${UPGRADE_DONE}  $CD_NAME ($OCP_CURRENT_VERSION)\n"
+                fi
+            fi
+        fi
+    done
+done
+
+rm -rf $TMP_DIR
+
+if [ ! -z $OCP_VERSION_FROM ];
```

Comment:
```suggestion
if [ -n $OCP_VERSION_FROM ];
```
As before, `! -z` == `-n`, and this will always be true.